shared/nav.desktop.js

```diff
@@ -163,6 +164,10 @@ class Nav extends Component<void, Props, State> {
   }
 
   shouldComponentUpdate (nextProps, nextState) {
+    if (this.props.folderBadge !== nextProps.folderBadge) {
```

Comment:
any 'i changed a prop and the thing didn't render' type problem is likely caused by this. the opposite (i changed nothing related to this component and it rendered) is likely noticable using the react perf tools wasted stuff https://facebook.github.io/react/docs/perf.html